samples/amqp/receiver/main.go

```diff
@@ -30,17 +30,22 @@ func sampleConfig() (server, node string, opts []ceamqp.Option) {
 	if err != nil {
 		log.Fatal(err)
 	}
+
 	if u.User != nil {
 		user := u.User.Username()
 		pass, _ := u.User.Password()
-		opts = append(opts, ceamqp.WithConnOpt(amqp.ConnSASLPlain(user, pass)))
+
+		opts = &amqp.ConnOptions{
+			SASLType: amqp.SASLTypePlain(user, pass),
+		}
 	}
 	return env, strings.TrimPrefix(u.Path, "/"), opts
 }
 
 func main() {
 	host, node, opts := sampleConfig()
-	p, err := ceamqp.NewProtocol(host, node, []amqp.ConnOption{}, []amqp.SessionOption{}, opts...)
+	p, err := ceamqp.NewProtocol(context.Background(), host, node, opts, &amqp.SessionOptions{},
+		&amqp.SenderOptions{}, &amqp.ReceiverOptions{})
```

Comment:
would be much nicer if those would not be pointers IMHO

Reply:
made it use the struct instead of pointers. thanks